src/coreclr/jit/fginline.cpp

```diff
@@ -590,14 +590,16 @@ class SubstitutePlaceholdersAndDevirtualizeWalker : public GenTreeVisitor<Substi
 #endif // DEBUG
 
                 CORINFO_CONTEXT_HANDLE context                = nullptr;
+                InlineContext*         inlinersContext        = m_compiler->compInlineContext;
```

Comment:
Do we ever succeed in devirtualizing w/o having the late devirt info?

Because if we do we'll still possibly be using the wrong context for the inline.

Reply:
I think this can be set to `nullptr` now, I can add an assertion `inlinersContext != nullptr`.

Reply:
Given that now we need to save the context for virtual calls anyway, we can assert `(call->gtCallMoreFlags & GTF_CALL_M_HAS_LATE_DEVIRT_INFO) != 0`